### ISSUE TITLE
Deeplab on mobile performace

### ISSUE THREAD

**Issue by ilous12** [tensorflow/models#6464]
Hi @aquariusjay I tried to improve inference speed on android and used mobilenetv2.

I set output_stride = 32 to remove the SpaceToBatchNd,BatchToSpaceNd in deeplab and I got a fastest inference speed.
But The accuracy was more lower.

I have 2 questions

(1) Is it possible "output_stride == 16 without SpaceToBatchNd,BatchToSpaceNd"?
(2) What do I set dilation values to 1?

**Comment by aquariusjay** [comment]
Hi ilous12,

May I ask how much lower?
If I remember correctly, it should drop by less than 2% mIOU.
You could also pretrain the model on COCO dataset to get better performance.

Regarding your questions:
You could (1) use kernel size 5x5 in those expanded_conv (which has the same receptive as 3x3 expnded_conv with rate = 2), and (2) then set last stride from 2 to 1.

Hope that helps.

**Comment by ilous12** [comment]
thanks @aquariusjay 

I understood you said to change belows.

slim/mobilenet_v2.py

```
V2_DEF = dict(
    defaults={
        # Note: these parameters of batch norm affect the architecture                                                                                                                                                                                                                           
        # that's why they are here and not in training_scope.                                                                                                                                                                                                                                    
        (slim.batch_norm,): {'center': True, 'scale': True},
        (slim.conv2d, slim.fully_connected, slim.separable_conv2d): {
            'normalizer_fn': slim.batch_norm, 'activation_fn': tf.nn.relu6
        },
        (ops.expanded_conv,): {
            'expansion_size': expand_input(6),
            'split_expansion': 1,
            'normalizer_fn': slim.batch_norm,
            'residual': True
        },
        (slim.conv2d, slim.separable_conv2d): {'padding': 'SAME'}
    },
    spec=[
        op(slim.conv2d, stride=2, num_outputs=32, kernel_size=[3, 3]),
        op(ops.expanded_conv,
           expansion_size=expand_input(1, divisible_by=1),
           num_outputs=16),
        op(ops.expanded_conv, stride=2, num_outputs=24),
        op(ops.expanded_conv, stride=1, num_outputs=24),
        op(ops.expanded_conv, stride=2, num_outputs=32),
        op(ops.expanded_conv, stride=1, num_outputs=32),
        op(ops.expanded_conv, stride=1, num_outputs=32),
        op(ops.expanded_conv, stride=2, num_outputs=64),
        op(ops.expanded_conv, stride=1, num_outputs=64),
        op(ops.expanded_conv, stride=1, num_outputs=64),
        op(ops.expanded_conv, stride=1, num_outputs=64),
        op(ops.expanded_conv, stride=1, num_outputs=96),
        op(ops.expanded_conv, stride=1, num_outputs=96),
        op(ops.expanded_conv, stride=1, num_outputs=96),
        op(ops.expanded_conv, stride=2, num_outputs=160),
        op(ops.expanded_conv, stride=1, num_outputs=160),
        op(ops.expanded_conv, stride=1, num_outputs=160),
        op(ops.expanded_conv, stride=1, num_outputs=320),
        op(slim.conv2d, stride=1, kernel_size=[1, 1], num_outputs=1280)
    ],
)
```

Can you give codes what do I change?

**Comment by hoonkai** [comment]
Hi Can I ask why setting `output_stride=32` removes `SpaceToBatchNd`?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
